src/modular/boxed_residue/mul.rs

```diff
@@ -208,15 +211,17 @@ pub(crate) fn montgomery_mul(z: &mut [Word], x: &[Word], y: &[Word], m: &[Word],
         let cx = c.wrapping_add(c2);
         let cy = cx.wrapping_add(c3);
         z[n + i] = cy;
+
+        // TODO(tarcieri): eliminate data-dependent branches
```

Comment:
I left this to handle in a followup because the performance hit seems substantial (~17%), whereas this PR itself seems to have only a slight (~1%) impact.